### PR TITLE
Use tailwindcss

### PR DESCRIPTION
**Pull request by dgdavid**
## TL;DR

Set up and start using [tailwindcss](https://tailwindcss.com/) as UI framework.

## The long version

So far, we had not decided which UI/CSS approach/toolkit use in the React branch. Instead, apart from [those experiments](https://github.com/jangouts/jangouts/pull/266) to test how to offer [dynamic layouts](https://github.com/jangouts/jangouts/pull/232), we were just adding rough CSS rules in plain *.css files.

It's not an easy decision to choose among so many (fun and great) ways to build the user interface in the Web development world. [Styled components](https://github.com/styled-components/styled-components)? [emotion](https://github.com/emotion-js/emotion)? [goober](https://github.com/cristianbote/goober)? [css-modules](https://github.com/css-modules/css-modules)? [stylable](https://github.com/wix/stylable)? whatever other [CSS in JS](https://github.com/MicheleBertoli/css-in-js) library?

Honestly, I haven't had the opportunity to work with any of those libraries to have enough knowledge and choose the _right_ one for this project.

But the time has arrived and we must stop just _prototyping_ and start working with a reliable UI toolkit. And that's why I think that the simple and straightforward could continue using _plain CSS_, but using the well known utility-first framework [tailwindcss](https://tailwindcss.com/).

To not extend too much this (already long) PR description, let me use a couple of links to justify the chose

* https://tailwindcss.com/#what-is-tailwind
* https://tailwindcss.com/docs/utility-first/#app

   > Now I know what you're thinking, _"this is an atrocity, what a horrible mess!"_ and you're right, it's kind of ugly. In fact it's just about impossible to think this is a good idea the first time you see it — **you have to actually try it**.
* https://www.smashingmagazine.com/2020/02/tailwindcss-react-project/

* Further readings (long and a bit old articles)
  
  * https://frontstuff.io/in-defense-of-utility-first-css 
  * https://adamwathan.me/css-utility-classes-and-separation-of-concerns/ (a very long and old one)

## Thus,

this PR set up and start using _tailwindcss_ by styling the login form. For sure, I've tried to follow its core concepts (i.e., respecting the mobile-first breakpoint system), but there is room yet to still applying [their recommended best practices](https://tailwindcss.com/docs/controlling-file-size/) (i.e., [removes not use colors](https://tailwindcss.com/docs/controlling-file-size/#limiting-your-color-palette), [removes not needed breakpoints](https://tailwindcss.com/docs/controlling-file-size/#removing-unused-breakpoints), etc).

## Thoughts?

Does it look good as starting point for you? Or maybe are you thinking that [it's just tomorrow's Bootstrap](https://blog.logrocket.com/tailwind-css-is-it-tomorrows-bootstrap-ebe560f9d00b/)?

## Screenshots

<details>
<summary>Click to show/hide screenshots</summary>

---

* _Classic_ theme

| Portrait | Landscape |
|-|-|
| Screenshot_2020-04-05 Jangouts(6) |  Screenshot_2020-04-05 Jangouts(7) |

* _Green_ theme

| Portrait | Landscape |
|-|-|
| Screenshot_2020-04-05 Jangouts(10) | Screenshot_2020-04-05 Jangouts(9) |

</details>

## Resources

Apart from its really nice documentation

* https://tailwindcss.com/docs

there are plenty of resources for/around _tailwindcss_

* https://tailwindcss.com/resources
* https://github.com/aniftyco/awesome-tailwindcss
* :tv: https://www.youtube.com/watch?v=nqNIy8HkEQ8 - Tailwind CSS Tips, Tricks & Best Practices
*  ...

### CODE REVIEW

File: src/components/LoginForm/LoginForm.js [hunk 1]
```diff
@@ -82,9 +82,9 @@ function renderError(error) {
 function LoginForm() {
   const dispatch = useDispatch();
   const [rooms, setRooms] = useState([]);
+  const [selectedRoom, setSelectedRoom] = useState({});
```
Comment (dgdavid) [inline src/components/LoginForm/LoginForm.js:85]: @imobachgs, as you worked around here recently, I need your insights here, please.

The thing is that I've changed a bit the layout to share the same _row_ for the PIN field and the submit action. It can be argued that this only complicates the markup... and it's true. However, the motivation behind such decision is to keep the form fully visible when using a small device in the landscape orientation. I mean, 


| Before | After |
|-|-|
| Screenshot_2020-04-04 Jangouts(1) | Screenshot_2020-04-05 Jangouts(8) |
| Screenshot_2020-04-05 Jangouts(1) | Screenshot_2020-04-07 Jangouts |

Just trying to follow one of the main core concepts of tailwind: [mobile first](https://tailwindcss.com/docs/responsive-design#mobile-first).

Thus, this commit (1ed597369ab2d3c40a70a56624f30fe0f1475fde) rid of the `watch` and use a hook to keep the selected room in the internal state. Do you see think this is right?

Thanks!

File: src/styles/themes.css [hunk 2]
```diff
@@ -0,0 +1,15 @@
+.theme-classic {
```
Comment (dgdavid) [inline src/styles/themes.css:1]: I'm using [CSS custom properties](https://developer.mozilla.org/en-US/docs/Web/CSS/--*) and a _root_ CSS class  as a simple mechanism to support theming.

Probably, it would be a better idea switching to something more elaborated like [tailwindcss-theming plugin](https://github.com/hawezo/tailwindcss-theming).

**Comment by ancorgs** [comment]
This pull request is my first contact with tailwindcss... and also with styled components, with emotion, with goober and all the other technologies you named. I fully trust your criteria regarding that choice.

**Comment by dgdavid** [comment]
@ancorgs 

> I fully trust your criteria regarding that choice.

Thank you so much, but take into account what I said

> I haven't had the opportunity to work with any of those libraries

It's true that I usually read a lot and tried to follow what is being cooked in the web development world, but nothing more. At this time it could be said that I'm disconnected from that area ;)

File: src/styles/themes.css (hunk 2 above)
Comment (ancorgs) [reply src/styles/themes.css:1]: Making it possible for the admin of the site to define which theme to use is a nice feature. I think the approach you followed is enough for that.

I don't think that going further (dark mode, allowing the client to change the theme, select a theme based on the client device, etc.) is important enough to deserve the introduction of another not-so-small dependency.
